Firestore/core/test/firebase/firestore/util/executor_test.cc

```diff
@@ -243,6 +244,60 @@ TEST_P(ExecutorTest, DuplicateTagsOnOperationsAreAllowed) {
   EXPECT_EQ(steps, "12");
 }
 
+TEST_P(ExecutorTest, ConcurrentExecutorsWork) {
+  /**
+   * A mix of a countdown latch and a barrier. All threads that bump the
+   * countdown block until the count becomes zero.
+   */
+  class BlockingCountdown {
+   public:
+    explicit BlockingCountdown(int threads) : threads_(threads) {
+    }
+
+    int count() const {
+      std::lock_guard<std::mutex> lock(mutex_);
+      return threads_;
+    }
+
+    /** Awaits the completion of all threads. */
+    void Await() {
+      std::unique_lock<std::mutex> lock(mutex_);
+      while (threads_ > 0) {
+        is_zero_.wait(lock);
+      }
+    }
+
+    /**
+     * Bumps the counter down by one and waits for the counter to become zero.
+     */
+    void Bump() {
+      std::unique_lock<std::mutex> lock(mutex_);
+
+      threads_ -= 1;
+
+      is_zero_.wait(lock, [this] { return threads_ == 0; });
```

Comment:
Is this to ensure concurrency (guarantee that `Bump`s cannot all happen on a single thread)? Can you add a comment?

Reply:
Yes; the class comment tries to do this, but apparently not enough. Added a comment here too.